sections/multicolumn.liquid

```diff
@@ -80,18 +80,39 @@
                       style="padding-bottom: {{ 1 | divided_by: highest_ratio | times: 100 }}%;"
                     {% endif %}
                   >
-                    {%- capture sizes -%}
-                      (min-width: 990px) {% if section.blocks.size <= 2 %}710px{% else %}550px{% endif %}, (min-width:
-                      750px) {% if section.blocks.size == 1 %}710px{% else %}550px{% endif %}, calc(100vw - 30px)
+                    {%- liquid
+                      assign numberOfColumns = section.settings.columns_desktop
+                      assign numberOfColumnsMobile = section.settings.columns_mobile
+                      assign maxImageWidth = settings.page_width | times: 2
+                      if section.settings.image_width == 'half'
+                        assign imageWidth = 0.5
+                      elsif section.settings.image_width == 'third'
+                        assign imageWidth = 0.33
+                      else
+                        assign imageWidth = 1
+                      endif
+                    -%}
+                    {%- capture widths -%}
+                      {{ settings.page_width | times: imageWidth | divided_by: numberOfColumns | round: 0 | times: 2 }},
+                      {{ settings.page_width | times: imageWidth | divided_by: numberOfColumns | round: 0 }},
+                      {{ 989 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 989 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }},
+                      {{ 749 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 749 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
+                      {{ 374 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 374 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
+
                     {%- endcapture -%}
+                    {% capture sizes %}
+                      (min-width: 990px) {{ settings.page_width | times: imageWidth | divided_by: numberOfColumns | round: 0 }}px,
+                      (min-width: 750px) {{ 989 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}px,
+                      (min-width: 375px) {{ 749 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}px,
+                      {{ 375 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}px
```

Comment:
The only issue I see with this is that it will always jump to the next size... E.g. if the screen width is 375 it will use an image that is 749w (assuming imageWidth of 1).

I think you can replace these three lines with:
```
calc(100vw * {{ imageWidth | divided_by: numberOfColumnsMobile }})
```

Reply:
I think we can be more flexible/accurate here without much more complexity if we apply the same sort of math to css calculations replacing fixed values like `989` with `100vw`. This will essentially create a dynamic value that also factors in the viewport width, rather than just the max breakpoint width.

Also I think we often define 3 breakpoints and a non-specific mobile size that is essentially a max-width: 749px. I think if you make the change above you'll find you don't need the min-width 375 breakpoint
- `min-width: {{ settings.page_width}}` (image sizes won't vary above this breakpoint and can use a fixed sized like you have currently)
- `min-width: 990px` (image sizes vary with viewport between 990 and 1200)
- `min-width: 750px` (image sizes vary with viewport between 750 and 990)

Reply:
I decided to go with what Stu suggested 👆.